src/ng/location.js

```diff
@@ -48,17 +48,20 @@ function parseAppUrl(relativeUrl, locationObj) {
   }
 }
 
+function startsWith(haystack, needle) {
```

Comment:
I'd think a generic helper like this should. Could even use [String#startsWith](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/String/startsWith) when available?